### ISSUE TITLE
Intermittent test failures on FreeBSD

### ISSUE THREAD

**Issue by ararslan** [JuliaLang/julia#45668]
Most, but not all, tester jobs on FreeBSD have been failing with the following error. It seems to have started happening right after https://github.com/JuliaLang/julia/pull/45560 was merged. cc @vtjnash

```
Assertion failed: (!err), function jl_thread_suspend_and_get_state, file ./signals-unix.c, line 390.
signal (6): Abort trap
in expression starting at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/profile_spawnmany_exec.jl:14
Allocations: 479591 (Pool: 475168; Big: 4423); GC: 1
┌ Error: A "spawn and wait lots of tasks" test failed
│   n = 200000
│   proc.exitcode = 0
│   proc.termsignal = 6
│   success(proc) = false
│   timeout = true
└ @ Main.Test5Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:313
Assertion failed: (!err), function jl_thread_suspend_and_get_state, file ./signals-unix.c, line 390.
signal (6): Abort trap
in expression starting at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/profile_spawnmany_exec.jl:14
Allocations: 2935 (Pool: 2923; Big: 12); GC: 0
┌ Error: A "spawn and wait lots of tasks" test failed
│   n = 2000000
│   proc.exitcode = 0
│   proc.termsignal = 6
│   success(proc) = false
│   timeout = false
└ @ Main.Test5Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:313
threads                                  (1) |         failed at 2022-06-09T17:21:29.742
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:319
  Expression: success(proc)
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:320
  Expression: !timeout
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:319
  Expression: success(proc)
```

**Comment by vtjnash** [comment]
https://github.com/JuliaLang/julia/commit/416dba5b216f6290ba71cba40ff67981495fa822 ?

But this should only be reached if the machine is severely oversubscribed

**Comment by ararslan** [comment]
Ah yeah, that did fix the assertion failure as shown in the OP but the surrounding failures are still present:
```
┌ Error: A "spawn and wait lots of tasks" test failed
│   n = 20000
│   proc.exitcode = 0
│   proc.termsignal = 1
│   success(proc) = false
│   timeout = true
└ @ Main.Test64Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:313
┌ Warning: Terminating `profile_spawnmany_exec.jl` required signal 1
└ @ Main.Test64Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:298
┌ Error: A "spawn and wait lots of tasks" test failed
│   n = 200000
│   proc.exitcode = 0
│   proc.termsignal = 1
│   success(proc) = false
│   timeout = true
└ @ Main.Test64Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:313
┌ Warning: Terminating `profile_spawnmany_exec.jl` required signal 1
└ @ Main.Test64Main_threads /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:298
threads                                  (1) |         failed at 2022-06-10T15:45:21.862
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:319
  Expression: success(proc)
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:320
  Expression: !timeout
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:319
  Expression: success(proc)
Test Failed at /usr/home/julia/buildbot/w2_tester/tester_freebsd64/build/share/julia/test/threads.jl:320
  Expression: !timeout
```

Yesterday I rebooted all of the FreeBSD VMs on amdci6 that we use for both building and testing to see whether that might help and unfortunately it seems not to have.

**Comment by ararslan** [comment]
Fixed by #45694. Thanks Jameson!
